File: app_canibuythat/src/main/java/com/gb/canibuythat/screen/MainScreen.java
package com.gb.canibuythat.screen;

import com.gb.canibuythat.model.Balance;

public interface MainScreen extends Screen {
    void showChartScreen();

    void showLoginActivity();

    void setBalanceInfo(Balance balance);

    void showFilePickerActivity(String directory);

    void showBalanceUpdateDialog();

    void showEditorScreen(final Integer budgetItemId);
}

File: app_canibuythat/src/main/java/com/gb/canibuythat/screen/MainScreen.kt
package com.gb.canibuythat.screen

import com.gb.canibuythat.model.Balance
import com.gb.canibuythat.db.model.ApiSpending
import com.gb.canibuythat.ui.BalanceBreakdown

interface MainScreen : MonzoDispatchScreen {

    enum class SpendingsImportType {
        MONZO,
        NON_MONZO,
        ALL
    }

    fun showLoginActivity()

    fun setBalanceInfo(balance: Balance?)

    fun setTotalSaving(totalSaving: Double?)

    fun showPickerForImport(directory: String, spendingsImportType: SpendingsImportType)

    fun showBalanceUpdateDialog()

    fun showEditorScreen(spendingId: Int?)

    fun showDialog(title: String, text: String)

    fun setTitle(projectName: String?)

    fun showPickerForExport(suggestedPath: String)

    fun setProjectName(currentName: String?)

    fun showBalanceBreakdown(breakdown: BalanceBreakdown)

    fun sendFCMTokenToServer()
}
